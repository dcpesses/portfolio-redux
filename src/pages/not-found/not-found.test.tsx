import {render} from '@testing-library/react';
import NotFound from './index';

interface LinkProps {
  className: string,
  to: string,
  children: React.ReactNode
}

vi.mock('react-router-dom', () => {
  const reactRouterDom = vi.importActual('react-router-dom');
  return {
    ...reactRouterDom,
    Link: ({className, to, children}: LinkProps) => (
      <div className={className} data-to={to}>
        {children}
      </div>
    )
  };
});

describe('NotFound', () => {
  test('Should render as expected', () => {
    const {container} = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });
});
