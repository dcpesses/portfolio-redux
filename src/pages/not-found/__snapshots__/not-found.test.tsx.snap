// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`NotFound > Should render as expected 1`] = `
<div>
  <div
    class="not-found"
  >
    <main
      class="wrapper"
    >
      <h1>
        404
      </h1>
      <p>
        Sorry, this page does not exist.
      </p>
      <div>
        <div
          class="NotFound-link"
          data-to="/"
        >
          ‚Üê Go Home
        </div>
      </div>
    </main>
  </div>
</div>
`;
